src/main/kotlin/ru/tinkoff/fintech/kotlin/collections/LinkedQueue.kt

```
@@ -0,0 +1,62 @@
+package ru.tinkoff.fintech.kotlin.collections
+
+class LinkedQueue<T> : Queue<T> {
+    class Node<T>(var element: T? = null, var prev: Node<T>? = null)
+
+    private var head = Node<T>()
+    private var tail = head
+    var size = 0
+        private set
+
+    override fun offer(element: T): Boolean {
+        try {
+            tail.element = element
+            tail.prev = Node()
+            tail = tail.prev!!
+            size++
+            return true
+        } catch (e: Exception) {
+            return false
+        }
+    }
+
+    override fun remove(): T {
+        if (isEmpty()) {
+            throw(java.util.NoSuchElementException())
+        }
+        val result = head.element
```

лучше здесь добавь !!, а в ретурне убери

если ты знаешь что оно тебе понадобится, проверяй сразу, не надо делать лишнюю работу, а потом падать - падай сразу, fail fast